recipes/mimalloc/all/conanfile.py

```diff
@@ -203,3 +217,8 @@ def package_info(self):
                 self.cpp_info.system_libs.extend(["psapi", "shell32", "user32", "bcrypt"])
             elif self.settings.os == "Linux":
                 self.cpp_info.system_libs.append("rt")
+
+        # Starting version 2.0 mimalloc installs the headers in a mimalloc-2.0 subfolder.
```

Comment:
This is the most concerning part.
I'm exposing a path that shouldn't be exposed to make preserve 1.x compatibility on the test package.
1.x installs the headers loose at `include` whereas 2.x installs them at `include/mimalloc-2.0`. By exposing both paths the consumer is able to use the new path but also the older one.

Reply:
If the only reason is test_package compatibility, perhaps we need to rewrite its headers.
e.g.
```c
#if MIMALLOC_MAJOR_VERSION >= 2
#include "mimalloc-2.0/mimalloc.h"
#else
#include "mimalloc.h"
#endif
```

Reply:
I take this back. 
The headers do get installed into `include/mimalloc-2.0`, but then they include each other with no `mimaloc-2.0` prefix.
E.g.
`include/mimalloc-2.0/mimalloc-override.h` has this line in it: `#include <mimalloc.h>`. So the only `includedir` to be exposed is `include/mimalloc-2.0`.

Reply:
Starting version 1.7.0 they added this CMake option that disables all this subfolders nonsense. Enabling it makes the package layout "standard".
https://github.com/microsoft/mimalloc/blob/v1.7.0/CMakeLists.txt#L25